src/Microsoft.AspNet.Mvc.TagHelpers/Internal/AttributeMatcher.cs

```diff
@@ -114,9 +120,9 @@ public static class AttributeMatcher
 
         private class PresentMissingAttributes
         {
-            public IEnumerable<string> Present { get; set; }
+            public IList<string> Present { get; set; }
 
-            public IEnumerable<string> Missing { get; set; }
+            public IList<string> Missing { get; set; }
```

Comment:
What future changes? I expect future simplifications and removals, not additions.